### PR TITLE
fix: flaky vm create test

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
fix: flaky vm create test

the test failed a lot of time due to already existing datavolume. This was caused by a string which changed during test run and all test runs had the same string. This PRs moves the test to a
already existing context, where the string is set and do not change during test.

**Release note**:
```
NONE

```

### CODE REVIEW

File: modules/tests/test/create_vm_common_test.go [hunk 1]
```diff
@@ -162,144 +162,146 @@ var _ = Describe("Create VM", func() {
...
-					CDRom: &kubevirtv1.CDRomTarget{Bus: "sata"},
-				},
-			}
-			vmDisk2 := kubevirtv1.Disk{
-				Name: datavolumes[1].DiskName,
-				DiskDevice: kubevirtv1.DiskDevice{
-					Disk: &kubevirtv1.DiskTarget{Bus: "virtio"},
-				},
-			}
+			It("VM with attached PVCs/DVs and existing disks/volumes is created successfully, mode: "+string(c), func() {
```
Comment (0xFelix) [inline modules/tests/test/create_vm_common_test.go:165]: Use either `c` or `createMode`? Both are the same?

Comment (ksimon1) [reply modules/tests/test/create_vm_common_test.go]: fixed

File: modules/tests/test/create_vm_common_test.go [hunk 2]
```diff
@@ -162,144 +162,146 @@ var _ = Describe("Create VM", func() {
...
+			It("VM with attached PVCs/DVs and existing disks/volumes is created successfully, mode: "+string(createMode), func() {
+				mode := "manifest-mode"
 
-			// disk disk3 should be created by the task
+				if createMode == CreateVMTemplateMode {
+					mode = "template-mode"
+				}
 
-			// volume disk1 should be created by the task
+				datavolumes := []*datavolume.TestDataVolume{
```
Comment (0xFelix) [inline modules/tests/test/create_vm_common_test.go:172]: What do you think of using random names for the DVs? IIUC duplicated names were an issue?

File: modules/tests/test/create_vm_common_test.go [hunk 3]
```diff
@@ -162,144 +162,146 @@ var _ = Describe("Create VM", func() {
...
-			Expect(vm.Spec.Template.Spec.Volumes).To(ConsistOf(expectedVM.Spec.Template.Spec.Volumes))
-			Expect(vm.Spec.Template.Spec.Domain.Devices.Disks).To(ConsistOf(expectedVM.Spec.Template.Spec.Domain.Devices.Disks))
-			Expect(vm.Spec.Template.Spec.Domain.Devices.Disks[2].CDRom.Bus).To(Equal(kubevirtv1.DiskBusSATA))
+				vm, err := vm.WaitForVM(f.KubevirtClient, expectedVM.Namespace, expectedVM.Name,
+					"", config.GetTaskRunTimeout(), false)
+				Expect(err).ShouldNot(HaveOccurred())
+				// check all disks are present
+				Expect(vm.Spec.Template.Spec.Volumes).To(ConsistOf(expectedVM.Spec.Template.Spec.Volumes))
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Disks).To(ConsistOf(expectedVM.Spec.Template.Spec.Domain.Devices.Disks))
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Disks[2].CDRom.Bus).To(Equal(kubevirtv1.DiskBusSATA))
```
Comment (0xFelix) [inline modules/tests/test/create_vm_common_test.go:303]: Add more assertions like length of volumes slice, disk bus of other disks, ...?

File: modules/tests/test/create_vm_common_test.go [hunk 4]
```diff
@@ -28,278 +28,264 @@ var _ = Describe("Create VM", func() {
...
-				},
-			}
+		vm, err := vm.WaitForVM(f.KubevirtClient, expectedVM.Namespace, expectedVM.Name,
+			"", config.GetTaskRunTimeout(), false)
+		Expect(err).ShouldNot(HaveOccurred())
+		// check all disks are present
+		Expect(vm.Spec.Template.Spec.Volumes).To(ConsistOf(expectedVM.Spec.Template.Spec.Volumes))
+		Expect(vm.Spec.Template.Spec.Domain.Devices.Disks).To(ConsistOf(expectedVM.Spec.Template.Spec.Domain.Devices.Disks))
+	},
+		Entry("owned-dv=1", CreateVMTemplateMode, 0, map[datavolume.TestDataVolumeAttachmentType]int{datavolume.OwnedDV: 1}),
```
Comment (0xFelix) [inline modules/tests/test/create_vm_common_test.go:136]: Can you define the maps outside the entries in a var block? For readability?

**Comment by 0xFelix** [comment]
/lgtm

Thanks!

**Pull request merged**